src/dbus/server/dbus_agent.cpp

```diff
@@ -49,47 +51,58 @@ DBusAgent::DBusAgent(otbr::Ncp::ControllerOpenThread &aNcp)
 
 void DBusAgent::Init(void)
 {
-    DBusError   dbusError;
-    otbrError   error = OTBR_ERROR_NONE;
-    int         requestReply;
-    std::string serverName = OTBR_DBUS_SERVER_PREFIX + mInterfaceName;
+    otbrError error = OTBR_ERROR_NONE;
 
-    dbus_error_init(&dbusError);
-    DBusConnection *conn                = nullptr;
-    auto            connection_deadline = std::chrono::steady_clock::now() + std::chrono::seconds(30);
+    auto connection_deadline = Clock::now() + kDBusWaitAllowance;
 
-    while (true)
+    while ((mConnection = PrepareDBusConnection()) == nullptr && Clock::now() < connection_deadline)
     {
-        conn = dbus_bus_get(DBUS_BUS_SYSTEM, &dbusError);
-        if (conn != nullptr || std::chrono::steady_clock::now() > connection_deadline)
-        {
-            break;
-        }
-        otbrLogWarning("Failed to get DBus connection: %s: %s, will retry after 3 seconds", dbusError.name,
-                       dbusError.message);
-        dbus_error_free(&dbusError);
-        sleep(3);
+        otbrLogWarning("Failed to setup DBus connection, will retry after 1 second");
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+        mConnection = PrepareDBusConnection();
```

Comment:
We will call `PrepareDBusConnection()` again at the top of the while-loop. Is that intended?

Reply:
Good catch! Failed to remove the call below.

Reply:
Another thought is that, thanks to `unique_ptr`, even such mistake won't cause DBus connection leak.